src/mysql_protocol.erl

```diff
@@ -1220,6 +1257,33 @@ parse_auth_method_switch(AMSData) ->
        auth_plugin_data = AuthPluginData
       }.
 
+-spec parse_auth_more_data(binary(), binary()) -> #auth_more_data{}.
+parse_auth_more_data(?authmethod_caching_sha2_password, <<3>>) ->
+    %% With caching_sha2_password authentication, a single 0x03
+    %% byte signals Fast Auth Success.
+    #auth_more_data{message = fast_auth_success};
+parse_auth_more_data(?authmethod_caching_sha2_password, <<4>>) ->
+    %% With caching_sha2_password authentication, a single 0x04
+    %% byte signals a Full Auth Request.
+    #auth_more_data{message = full_auth_request};
+parse_auth_more_data(?authmethod_caching_sha2_password, Data) ->
+    %% With caching_sha2_password authentication, anything
+    %% other than the above should be the public key of the
+    %% server.
+    PubKey = case public_key:pem_decode(Data) of
+        [PemEntry = #'SubjectPublicKeyInfo'{}] ->
+            public_key:pem_entry_decode(PemEntry);
+        [PemEntry = #'RSAPublicKey'{}] ->
+            PemEntry
+    end,
+    #auth_more_data{message = {public_key, PubKey}};
+parse_auth_more_data(?authmethod_sha256_password, Data) ->
+    %% sha256_password authentication is superseded by
+    %% caching_sha2_password.
```

Comment:
... but we implement it anyway, because it's easy?

Reply:
Hm, partially ;) From the noncommittal comment on MySQLs website, I read this as "anything required for sha256_password can be done with the mechanisms included in caching_sha2_password". When experimenting, I sometimes received auth switches to sha256_password (though neither the server nor the user had it specified), so it looks like we must support it. This is not conclusive, repeating the very same at a later time didn't use sha256_password. In short, I don't know what is required to show up =^^=

Reply:
Good then. :-)